brand/schema.py

```diff
@@ -425,43 +425,62 @@ def resolve_harvest_data(self, info, tag, **kwargs):
             logger.error(f"Unexpected error resolving harvest data for {tag}: {str(e)}")
             return
 
-    def resolve_all_harvest_data(self, info, **kwargs):
-        commentary_queries = CommentaryModel.objects.all()
-        tag_feature_json_dict = {
-            query.brand.tag: query.feature_json for query in commentary_queries
-        }
+    def resolve_filtered_harvest_data(
+        self,
+        info,
+        show_on_sustainable_banks_page=False,
+        country=None,
+        state_licensed=None,
+        state_physical_branch=None,
+        **kwargs,
+    ):
+        """
+        This query is built specifically for the need of bank.green's sustainable-eco-banks page
+        Returns
+        """
+
+        commentaries = CommentaryModel.objects.all()
+
+        if show_on_sustainable_banks_page:
+            commentaries = commentaries.filter(
+                show_on_sustainable_banks_page=show_on_sustainable_banks_page
+            )
 
-        filtered_data = []
+        if country:
+            commentaries = commentaries.filter(brand__countries=country)
 
-        try:
-            features_field = next(
-                (
-                    field
-                    for field in info.field_nodes[0].selection_set.selections
-                    if field.name.value == "features"
-                ),
-                None,
+        if state_licensed:
+            state = StateModel.objects.get(tag=state_licensed)
+            commentaries = commentaries.filter(brand__state_licensed=state)
+
+        if state_physical_branch:
+            state = StateModel.objects.get(tag=state_physical_branch)
+            commentaries = commentaries.filter(brand__state_physical_branch=state)
+
+        requested_fields = kwargs.items()
+
+        def brand_has_requested_features(brand):
```

Comment:
It feels kind of weird to have a sub function in there, but I did manually test and it works fine. This might be a good to unittest in the future.